apps/channels/models.py

```diff
@@ -22,22 +22,20 @@
 TELEGRAM = "telegram"
 WHATSAPP = "whatsapp"
 FACEBOOK = "facebook"
+IN_APP = "in_app"
 
 
 class ChannelPlatform(models.TextChoices):
     TELEGRAM = "telegram", "Telegram"
     WEB = "web", "Web"
     WHATSAPP = "whatsapp", "WhatsApp"
     FACEBOOK = "facebook", "Facebook"
+    IN_APP = "in_app", "In App"
```

Comment:
> While SureAdhere is one of the in-app messaging platforms we currently support, what if we decide to add another provider in the future that also offers in-app messaging capabilities?

I'm not sure I quite follow this. If SA adds support for another provider e.g. telegram then OCS will still want to use the SA channel and SA will decide how to deliver that message to the user based on what provider that user is using to communicate with. But OCS won't know about the different downstream providers will it?

Reply:
I feel like we're missing eachother somewhere and maybe we should go to voice but I'll take one more go at explaining what I think.

When you say "in app", do you mean "communication through a mobile app"? Is that not what WhatsApp and Telegram are already?

In terms of this data model, a channel type should be 1-1 with how the user access it e.g. whatsapp vs telegram vs sureadhere

The `MessagingProvider` ties the channels to a way of sending and receiving messages and is 1-many since the same provider could facilitate communications on multiple channels and multiple providers can support the same channel.

If we take WhatsApp as an example, both Turn and Twilio provide WhatsApp messaging capabilities but which one the user chooses depends on which provider they are using.

If you still have doubts, let's schedule a call.